packages/arb-token-bridge-ui/src/components/RecoverFunds.tsx

```diff
@@ -0,0 +1,530 @@
+        }
+      })
+
+      const balances = await Promise.all(balancePromises)
+      // Aliased account will always have some leftover, we can't check for balance of 0, as it would always return 0
+      // We compare with 0.005 instead
+      return balances.filter(([, balance]) => balance.gt(parseEther('0.005')))
+    }
+  )
+}
+
+const TokenColumn: TableCellRenderer = ({ rowData }) => {
+  const [, balance, nativeCurrency] = rowData
+
+  return (
+    <span className="flex h-12 items-center align-middle">
+      <SafeImage
+        src={nativeCurrency.logoUrl}
+        height={20}
+        width={20}
+        fallback={
+          <div className="flex h-4 w-4 items-center justify-center bg-gray-dark text-sm font-medium">
+            ?
+          </div>
+        }
+      />
+      <span className="ml-1 text-xs">
+        {formatAmount(balance, nativeCurrency)}
+      </span>
+    </span>
+  )
+}
+
+export function RecoverFunds() {
+  const { address } = useAccount()
+  const [{ sourceChain }] = useNetworks()
+  const { isTestnet } = isNetwork(sourceChain.id)
+  const { data: lostFunds } = useFundsOnAliasedAddress({
+    address,
+    isTestnet
+  })
+  const [dialogProps, openDialog] = useDialog2()
+
+  if (!address || !lostFunds || lostFunds.length === 0) {
+    return null
+  }
+
+  return (
+    <>
+      <DialogWrapper {...dialogProps} />
+      <NoteBox className="left-0 right-0 m-auto w-[600px]">
+        <div className="flex items-center">
+          <div>{`There are funds on ${getAliasedAddress(
+            address
+          )} (Alias of ${address}) to be recovered.`}</div>
+          <Button
+            variant="primary"
+            onClick={() => openDialog('recover_funds')}
+            className="ml-1"
+          >
+            Recover funds
+          </Button>
+        </div>
+      </NoteBox>
+    </>
+  )
+}
+
+const ActionColumn: TableCellRenderer = ({ rowData }) => {
+  const [chainId, balance]: [ChainId, BigNumber, NativeCurrency] = rowData
+  const { address } = useAccount()
+  const [isLoading, setIsLoading] = useState(false)
+  const [destinationAddress, setDestinationAddress] = useState<
+    string | undefined
+  >(undefined)
+  const { handleError } = useError()
+  const signer = useEthersSigner()
+  const childChainProvider = getProviderForChainId(chainId)
+  const [{ sourceChain }] = useNetworks()
+  const { isTestnet } = isNetwork(sourceChain.id)
+  const { mutate } = useFundsOnAliasedAddress({
+    address,
+    isTestnet
+  })
+  const parentChainProvider = useMemo(() => {
+    const chains = getChains()
+    const chain = chains.find(chain => chain.chainId === chainId)!
+    return getProviderForChainId(
+      'parentChainId' in chain ? chain.parentChainId : 1
+    )
+  }, [chainId])
+
+  if (!address) {
+    return null
+  }
+
+  const inboxAddress = isNetwork(chainId).isOrbitChain
+    ? getInboxAddressFromOrbitChainId(chainId)
+    : getArbitrumNetwork(chainId).ethBridge.inbox
+
+  return (
```

Comment:
I think since the whole dialog is outside the context of the TransferPanel's move funds flow, users can use this without accepting the Terms of Service checkbox. I think we should reuse that component somewhere here to stay clear, legally. (if the ToS is not accepted yet)

Reply:
Will check with Val, previously user didn't have to agree to ToS for retryable dashboard